### PR TITLE
Adds Flow.asFlowWithContext

### PR DESCRIPTION
**Pull request by skyluc**
Adds `Flow.asFlowWithContext()` function.
It allows to have symmetry with the existing `FlowWithContext.asFlow()`, `Source.asSourceWithContext()` and `SourceWithContext.asSource` functions (being renamed in #26353).

The main purpose of this function is to reduce the boilerplate code needed to integrate with a flow which is using some context information, but which is not using `FlowWithContext`.

### CODE REVIEW

**Comment by patriknw** [comment]
OK TO TEST

File: akka-docs/src/main/paradox/stream/operators/Flow/asFlowWithContext.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Flow.asFlowWithContext
+
+Turns a Folw into a FlowWithContext which can propagate a context per element along a stream.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/stream/operators/Flow/asFlowWithContext.md]: typo Folw

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-docs/src/main/paradox/stream/operators/Flow/asFlowWithContext.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+## Signature
+
+@@signature [Flow.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala) { #asFlowWithContext }
+
+@@@
+
+## Description
+
+Turns a Flow into a FlowWithContext which can propagate a context per element along a stream.
+The first function passed into asFlowWithContext must turn incoming elements and contexts into elements of the flow, for every incoming elements.
```
Comment (hseeberger) [inline akka-docs/src/main/paradox/stream/operators/Flow/asFlowWithContext.md]: Maybe rephrase "must turn each incoming pair of element and context value into an element of this Flow"

File: akka-docs/src/main/paradox/stream/operators/Flow/asFlowWithContext.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+The second function passed into asFlowWithContext must turn elements into contexts, one context for every element.
```
Comment (hseeberger) [inline akka-docs/src/main/paradox/stream/operators/Flow/asFlowWithContext.md]: Maybe rephrase as "must turn each outgoing element of this Flow into an outgoing context value"

**Review by hseeberger — CHANGES_REQUESTED** [review]
Nice feature! I have two comments hopefully improving the wording of the docs.

**Comment by skyluc** [comment]
Updated for #26419

Thanks @hseeberger for the comments.

**Review by johanandren — APPROVED** [review]
LGTM, I must have missed reviewing this before somehow, sorry!
